### PR TITLE
Prevent fooling PerformanceObserver.supportedEntryTypes

### LINKED ISSUE #117: supportedEntryTypes API has some issues
The two (interrelated) problems are:

1) It's an attribute that is returning a new value each time.  This is a bad enough antipattern, known for long enough, that Web IDL tries to prevent it; it looks like the definition of this API used `FrozenArray` instead of sequence to try to work around that prevention.

2) The workaround is broken: the steps for the attribute getter return an infra list, but per Web IDL if you are returning a value of type `FrozenArray` you need to return an actual frozen Array object.  There is no implicit conversion from _list_ defined at https://heycam.github.io/webidl/#idl-frozen-array or https://heycam.github.io/webidl/#es-frozen-array or https://infra.spec.whatwg.org/#lists (compare to sequence, which _does_ define such a conversion at https://heycam.github.io/webidl/#idl-sequence

If we want to return a new object each time here, this should be a method, not an attribute.  If we want to return the same object each time, then using a FrozenArray makes sense, but it's worth thinking about where that object is going to be cached, because it needs to be cached somewhere.  Probably in a slot on either the PerformanceObserver interface object or the global.

My personal preference would be for this API to be a static method returning a sequence, fwiw.

@npm1 @domenic @foolip @yoavweiss @igrigorik @spanicker @bakulf @w3c/web-assembly 

P.S. I am aware that Chrome has already shipped this, with the "return a new value each time" behavior.  I am fairly opposed to Gecko shipping anything of the sort.

### PR DESCRIPTION
**Pull request by plehegar**
This intended to help resolve https://github.com/w3c/performance-timeline/issues/117

### CODE REVIEW

**Comment by plehegar** [comment]
This test guarantees that official entries claimed to be supported by an implementation are actually supported. It repeats checks from respective specifications/tests providing those definitions.

More precisely, it follows:
 
https://github.com/web-platform-tests/wpt/blob/master/navigation-timing/supported_navigation_type.window.js
https://github.com/web-platform-tests/wpt/blob/master/resource-timing/supported_resource_type.any.js
https://github.com/web-platform-tests/wpt/blob/master/user-timing/supported-usertiming-types.any.js
https://github.com/web-platform-tests/wpt/blob/master/paint-timing/supported-paint-type.window.js
https://github.com/web-platform-tests/wpt/blob/master/longtask-timing/supported-longtask-types.window.js

NOTE: I probably need to rewrite my test with .window.js and .any.js ...

File: timing-entrytypes-registry/registry.html [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+<head>
+<meta charset="utf-8" />
+<title>Registry test for Performance Timeline</title>
+<link rel="help" href="https://w3c.github.io/timing-entrytypes-registry/">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+const STEPS = {};
+
+const types = (self.PerformanceObserver
```
Comment (npm1) [inline timing-entrytypes-registry/registry.html]: Can we also assert_unreached in the async_tests so we early abort if supportedEntryTypes is not defined?

File: timing-entrytypes-registry/registry.html [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+function test_support(def) {
+  if (!types || !types.includes(def[0])) {
+    return;
+  }
+  const desc = `'${def[0]}' entries should be observable`;
+  const t = async_test(desc);
+
+  STEPS[def[0]] = (entry) => {
+    t.step(() => assert_equals(Object.prototype.toString.call(entry),
```
Comment (npm1) [inline timing-entrytypes-registry/registry.html]: Hmm is there a cleaner way to do this?

**Review by npm1 — COMMENTED** [review]
Looks good other than the need to account for window/worker differences (this is fine as a window test though).

File: timing-entrytypes-registry/registry.html (hunk 1 above)
Comment (plehegar) [reply timing-entrytypes-registry/registry.html]: this is taken care of by  if (!types || !types.includes(def[0])) return;

File: timing-entrytypes-registry/registry.html (hunk 2 above)
Comment (plehegar) [reply timing-entrytypes-registry/registry.html]: it's actually pretty clean I think. It's listening to all supported entries by default so one doesn't have to worry about it when adding a new entry.

File: timing-entrytypes-registry/registry.any.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+// mark
+self.performance.mark('mymark');
+
+// measure
+self.performance.measure('mymeasure');
+
+// resource
+fetch(self.location.href + "?" + Math.random());
+
+// longtask
```
Comment (npm1) [inline timing-entrytypes-registry/registry.any.js]: I think for now longtasks are only supported in Window.

**Review by npm1 — APPROVED** [review]

**Comment by npm1** [comment]
Also look at Taskcluster failures before merging.

File: timing-entrytypes-registry/registry.any.js (hunk 3 above)
Comment (plehegar) [reply timing-entrytypes-registry/registry.any.js]: indeed

**Pull request merged**